datafusion/optimizer/src/decorrelate_predicate_subquery.rs

```diff
@@ -155,6 +152,17 @@ impl OptimizerRule for DecorrelatePredicateSubquery {
             _ => Ok(None),
         }
     }
+}
+
+impl OptimizerRule for DecorrelatePredicateSubquery {
+    fn try_optimize(
```

Comment:
cc @jackwener @alamb

Is this looking ok? The problem I'm encountering is that every queries optimization use the same shared AliasGenerator. Thus, the `alias ID` is ever increasing. Instead, I think it should be instantiated per query, which is what this code does.

Reply:
Nvm, apparently this is not ok since each optimization pass may be done multiple times.

Reply:
What do you think storing a reference to AliasGenerator in Optimizer so it can be reset during each call to `optimize`?

Reply:
Nvm, I think it could cause a potential race condition.

Reply:
I think it would be great to reset the alias generator at the start of each query (if that is not already done)